lib/portage/tests/ebuild/test_fetch.py

```diff
@@ -36,6 +38,72 @@
 
 
 class EbuildFetchTestCase(TestCase):
+
+    async def _test_interrupt(self, loop, server, async_fetch, pkg, ebuild_path):
+        """Test interrupt, with server responses temporarily paused."""
+        server.pause()
+        pr, pw = multiprocessing.Pipe(duplex=False)
+        timeout = loop.create_future()
+        loop.add_reader(pr.fileno(), lambda: timeout.done() or timeout.set_result(None))
+        self.assertEqual(
+            await async_fetch(
+                pkg,
+                ebuild_path,
+                timeout=timeout,
+                pre_exec=functools.partial(self._pre_exec_interrupt_patch, pw),
+            ),
+            -signal.SIGTERM,
+        )
+        loop.remove_reader(pr.fileno())
+        pw.close()
+
+        # Read pid written by _async_spawn_fetch_pre_wait hook (the
+        # corresponding write served to trigger the timeout above).
+        pid = pr.recv()
+
+        # Read pid written by _async_spawn_fetch_pre_terminate hook,
+        # in order to know when the ProcessLookupError test should
+        # succeed.
+        pid = pr.recv()
+        pr.close()
+
+        # Poll the process table until the pid has disappeared,
+        # and fail if a short timeout expires.
+        tries = 10
+        while tries:
+            tries -= 1
+            try:
+                self.assertRaises(ProcessLookupError, os.kill, pid, 0)
```

Comment:
This passes on my laptop, but not in CI. I wonder if the difference is something about pid 1 needing to reap zombies because the parent didn't wait on the process after it sent SIGTERM.

Reply:
Something is different about the wget behavior, and and apparently we have to follow up with a SIGKILL if SIGTERM does not cause immediate exit.